modules/dists/BlockDist.chpl

```diff
@@ -542,9 +549,20 @@ override proc Block.dsiNewRectangularDom(param rank: int, type idxType,
   if rank != this.rank then
     compilerError("Block domain rank does not match distribution's");
 
-  var dom = new unmanaged BlockDom(rank=rank, idxType=idxType, dist=_to_unmanaged(this),
-      stridable=stridable, sparseLayoutType=sparseLayoutType);
-  dom.dsiSetIndices(inds);
+  const whole = createWholeDomainForInds(rank, idxType, stridable, inds);
+
+  const dummyLBD = new unmanaged LocBlockDom(rank, idxType, stridable);
```

Comment:
This is an option. I suggest we do this later after considering the following argument.

Moving this code to an initializer does not feel to me like it would add clarity. Because to do so we would have to create TWO user-defined initializers for BlockDom, whereas currently we rely on the compiler-generated one.

There are two places in the code where a `BlockDom` object is created. Here it is created with `locDoms` array consisting of freshly-allocated objects. In the other place it is for privatization. There,`locDoms` is a copy of the original `BlockDom`.

So, the creation of `locDoms` cannot be factored out between these two cases. We would simply be moving code from next-to-`new`-expression into the newly-created initializer.

Reply:
It would feel clearer to me if the responsibility of creating LocBlockDom instances was given to BlockDom rather than Block.dsiNewRectangularDom. I won't hold up the PR over it, but I do think this is the right direction long term. I'd rather have two user-defined initializers with clear roles rather than relying on the compiler-generated initializer and putting initialization code before the ``new`` expression.